src/rcl_handle.cpp

```diff
@@ -85,8 +85,8 @@ NAN_METHOD(RclHandle::Dismiss) {
   info.GetReturnValue().Set(Nan::Undefined());
 }
 
-v8::Local<v8::Object> RclHandle::NewInstance(void* handle, RclHandle* parent,
-                                             std::function<int()> deleter) {
+v8::Local<v8::Object> RclHandle::NewInstance(
+    void* handle, RclHandle* parent, std::function<void(void*)> deleter) {
```

Comment:
Should we have the default deleter to `free` the pointer to simplify usage for the most common case?

On a side note, I recently found out that `free` on a c++ `new` allocated pointer is undefined behavior, similar for c++ `delete` on a `malloc` allocated pointer. This is another reason why moving the work of deallocation to the deleter function is a good idea, the old implementation will call `free` regardless of the pointer being `malloc` or `new` allocated.

Reply:
As you said, we cannot assume whether the `ptr` is allocated by `malloc` or `new`, so letting the caller decide seems reasonable. Although, we have to repeat `free` in most cases, any thoughts?